lib/facility.rb

```diff
@@ -1,14 +1,188 @@
+# Helper module to title case strings.
+module Helper
+  # Capitalizes the first letter of each word in a string
+  def self.title_case(input_string)
+    # Split the string into an array of words
+    words = input_string.split
+    # Map over the array of words and capitalize the first letter of each word
+    title_words = words.map do |word|
+      # If the word is in the exempt_words array, capitalize it
+      if exempt_words.include?(word.upcase)
+        word.upcase
+      # Otherwise, capitalize the first letter of the word
+      else
+        word.capitalize
+      end
+    end
+    # Join the array of words back into a string
+    title_words.join(' ')
+  end
+
+  def self.exempt_words
+    # List of words that should not be capitalized (states, DMV, road directions, etc.)
+    %w[DMV NY MD OR SE NE NW SW]
+  end
+end
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :website,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  # Initializes a facility
+  def initialize(data)
+    @name = data[:name]
+    @address = data[:address]
+    @phone = data[:phone]
+    @website = data[:website]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+  end
+
+  # Determines which state the facility is in and calls the appropriate method
+  def self.create_facility(state, facility_data)
+    # Switch statement to determine which state the facility is in
+    case state
+      # Calls the appropriate method based on the state
+    when 'OR' # Oregon
+      create_facility_from_oregon(facility_data)
+    when 'NY' # New York
+      create_facility_from_new_york(facility_data)
+    when 'MO' # Missouri
+      create_facility_from_missouri(facility_data)
+    else
+      # Returns an error if the state is not supported
+      raise "Unsupported state: #{state}"
+    end
+  end
+
+  # Creates a facility if the first argument in Facility.create_facility is 'OR'
+  def self.create_facility_from_oregon(facility_data)
+    # Map over the array of facility data and create a new Facility object for each facility
+    facility_data.map do |data|
```

Comment:
Yeah, I like that you're using a map here. The next thing you could do is once these are created, store them somewhere, maybe a class variable so that you only have to do this mapping once.